coffee_core/src/coffee.rs

```diff
@@ -352,10 +352,18 @@ impl PluginManager for CoffeeManager {
             .ok_or_else(|| error!("Repository with name: `{}` not found", repo))?;
 
         let status = repository.upgrade(&self.config.plugins, verbose).await?;
-        for plugins in status.plugins_effected.iter() {
-            self.remove(plugins).await?;
-            self.install(plugins, verbose, false).await?;
+
+        // if status is not up to date, we need to update the plugins as well
+        match status.status {
+            UpgradeStatus::UpToDate(_, _) => {}
+            _ => {
+                for plugins in status.plugins_effected.iter() {
+                    self.remove(plugins).await?;
+                    self.install(plugins, verbose, false).await?;
```

Comment:
```suggestion
                    self.install(plugins, verbose, verbose).await?;
```

We must pass down the verbose field

Reply:
This is `try_dynamic` 
`verbose` is already passed

Reply:
mh nice, we may want also thing about `try_dynamic`